shared/modules/transaction.utils.ts

```diff
@@ -121,49 +119,42 @@ export function parseStandardTokenTransactionData(data) {
   return undefined;
 }
 
-/**
- * Determines the contractCode of the transaction by analyzing the txParams.
- *
- * @param {object} txParams - Parameters for the transaction
- * @param {EthQuery} query - EthQuery instance
- * @returns {InferTransactionTypeResult}
- */
-export async function determineTransactionContractCode(txParams, query) {
-  const { to } = txParams;
-  const { contractCode } = await readAddressAsContract(query, to);
-  return contractCode;
-}
-
 /**
  * Determines the type of the transaction by analyzing the txParams.
  * This method will return one of the types defined in {@link TransactionType}
  * It will never return TRANSACTION_TYPE_CANCEL or TRANSACTION_TYPE_RETRY as these
  * represent specific events that we control from the extension and are added manually
  * at transaction creation.
  *
- * @param {object} txParams - Parameters for the transaction
- * @param {EthQuery} query - EthQuery instance
- * @returns {Promise<InferTransactionTypeResult>}
+ * @param txParams - Parameters for the transaction
+ * @param query - EthQuery instance
+ * @returns InferTransactionTypeResult
  */
-export async function determineTransactionType(txParams, query) {
+export async function determineTransactionType(
+  txParams: TransactionParams,
+  query: EthQuery,
+): Promise<InferTransactionTypeResult> {
   const { data, to } = txParams;
-  let result;
+  let result: TransactionType;
   let contractCode;
 
   if (data && !to) {
     result = TransactionType.deployContract;
   } else {
     const { contractCode: resultCode, isContractAddress } =
-      await readAddressAsContract(query, to);
+      await readAddressAsContract(query, to as string);
 
     contractCode = resultCode;
 
     if (isContractAddress) {
       const hasValue = txParams.value && Number(txParams.value) !== 0;
 
-      let name;
+      let name: string = '';
       try {
-        ({ name } = data && parseStandardTokenTransactionData(data));
+        const parsedData = data && parseStandardTokenTransactionData(data);
+        if (parsedData && parsedData.name) {
```

Comment:
Using `&&` to assert non-nullability like this is not a good pattern, since it pollutes the type signature of the assignee. 

In this case `parsedData` gets stuck with a `""` type which is neither useful nor actually representative of its expected behavior. 

Could we narrow `parsedData` down to `TransactionDescription | undefined`? That way we can just use optional chaining for the null check.

```suggestion
        const parsedData = data
          ? parseStandardTokenTransactionData(data)
          : undefined;
        if (parsedData?.name) {
```

Reply:
Yeah of course ! it is definitely a better type check !